### PR TITLE
cleanup scoreWDL.py

### PR DESCRIPTION
**Pull request by Disservin**
First cleanup patch of the python code. Group logic and remove usage of any "globals".
I am pretty open to changes, concering naming or the way I divided the functions.

This serves as a first step for later refactors.

### CODE REVIEW

**Comment by Disservin** [comment]
I just set the PR status to draft to wait for input and proper testing. So far `python scoreWDL.py` seems to run fine.

**Comment by vondele** [comment]
had a first look, the code style looks good to me. I've done a first check that it returns the same text output as master with all default arguments.

File: scoreWDL.py [hunk 1]
```diff
@@ -4,384 +4,491 @@
...
-            model_ms, poly3(model_ms, *popt_bs), "m-", label="fit: " + label_bs
+        if args.fit:
+            title = "Summary of win-draw-loss model analysis"
+            pgnName = "WDL_model_summary.png"
+        else:
+            title = "Summary of win-draw-loss data"
+            pgnName = f"WDL_data_{args.yData}.png"
+
+        plot = WdlPlot(
+            "Summary of win-draw-loss model analysis", "WDL_model_summary.png"
```
Comment (robertnurnberg) [inline scoreWDL.py:468]: to be consistent with master, this line should use `title` and `pgnName`

File: scoreWDL.py [hunk 2]
```diff
@@ -4,384 +4,491 @@
...
+        data_loader.load_json(),
+        args.moveMin,
+        args.moveMax,
+        args.NormalizeToPawnValue,
+        args.yData,
+    )
+
+    xs, ys, zwins, zdraws, zlosses = data_loader.convert_to_model(win, draw, loss)
+
+    wdl_model = WdlModel(args, plot)
```
Comment (robertnurnberg) [inline scoreWDL.py:485]: plot here could be undefined

maybe a better logic would be to make `plot` a member data of `WdlModel`

**Comment by robertnurnberg** [comment]
Looks very nice overall.

File: scoreWDL.py (hunk 1 above)
Comment (Disservin) [reply scoreWDL.py:468]: oh thanks, I didnt notice I hardcoded those :p

**Comment by robertnurnberg** [comment]
lgtm

**Pull request merged**